src/www/app/outline_server_repository/index.ts

```diff
@@ -100,7 +100,8 @@ export class OutlineServerRepository implements ServerRepository {
   add(accessKey: string) {
     this.validateAccessKey(accessKey);
 
-    const server = this.createServer(uuidv4(), accessKey, SHADOWSOCKS_URI.parse(accessKey).tag.data);
+    const server = this.createServer(uuidv4(), accessKey, this.inferAccessKeyName(accessKey));
```

Comment:
I'd rather not pass a name if it wasn't explicitly specified. Let the Server implementation do that. Otherwise there's no way to differentiate the implicit and explicit names.
Also, it's better if the "default name" logic lives in one place, rather than spread out.

Reply:
BTW, we are deviating from the existing logic that calls the server "Outline Server". That's usually ok for static servers because we show the IP address anyway, but it doesn't work for dynamic servers without a subtitle.

Perhaps we keep the same default name logic, but set the address subtitle to be the name you are creating here.

Reply:
Yeah generally I'm very confused about where the access key logic should go generally, it seems to be spread out all over. This is what my previous refactor attempted to address but it ended up going a different way. Let me think about this some more.

Reply:
I put all the default name logic in the Server itself - I'm not sure if that's what you meant. Take a look.